### PR TITLE
File and folder move

### PR DESCRIPTION
**Pull request by barrywark**
API for moving `File` or `Folder` from source `Folder` or `Project` to a destination `Folder` or `Project`.

### CODE REVIEW

File: src/ovation/handler.clj [hunk 1]
```diff
@@ -10,7 +10,7 @@
             [ovation.schema :refer :all]
             [ovation.logging]
             [ovation.routes :refer [router]]
-            [ovation.route-helpers :refer [annotation get-resources post-resources get-resource post-resource put-resource delete-resource rel-related relationships post-revisions* get-head-revisions*]]
+            [ovation.route-helpers :refer [annotation get-resources post-resources get-resource post-resource put-resource delete-resource rel-related relationships post-revisions* get-head-revisions* move-file*]]
```
Comment (rensm) [inline src/ovation/handler.clj]: Do you need to add move-folder as well?

Comment (barrywark) [inline src/ovation/handler.clj]: No, we use `move-file*` for both. I struggled with the name. Happy to take suggestions for renaming `move-file*` to something better. `move-file-or-folder*` would be ok, I guess but I tend to dislike "or" or "and" in a function name.

Comment (barrywark) [inline src/ovation/handler.clj]: Maybe `move-contents*`?

File: test/ovation/test/handler.clj [hunk 2]
```diff
@@ -513,35 +514,59 @@
...
+    (let [apikey TOKEN
+          id     (str (util/make-uuid))
+          body   {:source      (str (util/make-uuid))
+                  :destination (str (util/make-uuid))}
+          post   (mock-req (-> (mock/request :post (util/join-path ["" "api" ver/version "folders" id "move"]))
+                             (mock/body (json-post-body body))) apikey)
+          expected {:something "awesome"}]
+      (body-json post) => expected
+      (provided
+        (rh/move-file* anything id body) => expected))))
```
Comment (rensm) [inline test/ovation/test/handler.clj]: shouldn't this be rh/move-folder

**Pull request merged**